### ISSUE TITLE
database.batch does not retry aborted transactions

### ISSUE THREAD

**Issue by olavloite** [googleapis/python-spanner#1133]
The standard example for writing data with mutations uses `database.batch`: https://cloud.google.com/spanner/docs/getting-started/python#write-data-with-mutations

`database.batch` however does not automatically retry the transaction if it is aborted by Spanner. This causes errors if you try to use this method to insert a large amount of data, or if there are lock contentions on the data that you insert.

Either:
1. The sample(s) should be updated to show how to use `run_in_transaction` to use mutations.
2. And/or: `database.batch` should also automatically retry aborted transactions.

**Comment by aakashanandg** [comment]
Hi @olavloite, after reviewing the database.batch API, we found that commit calls within the batch are automatically retried with a default retry count of 5. For reference, you can check this line in the code: [link](https://github.com/googleapis/python-spanner/blob/main/google/cloud/spanner_v1/batch.py#L220). `batch.insert` API simply appends the values to the mutations list, and those values are persisted in the database when the commit API is called in the Spanner client.

**Comment by olavloite** [comment]
Yeah, this bug is specifically for `Aborted` errors. If the transaction is aborted by Spanner (meaning: Spanner returns an error with error code `Aborted`), then the transaction should be retried. The retry mechanism should be the same as for `run_in_transaction`; It should do a back-off and retry, using the back-off value that is included in the `Aborted` error. There should not be a maximum number of retries, instead it should stop retrying if the deadline has been exceeded.

**Comment by aakashanandg** [comment]
I understand. The run_in_transaction method currently evaluates the deadline value using the keyword arguments, as defined here: [link](https://github.com/googleapis/python-spanner/blob/a6811afefa6739caa20203048635d94f9b85c4c8/google/cloud/spanner_v1/session.py#L452). In my opinion, it's the client's responsibility to pass the timeout (in seconds) as part of the keyword arguments for the transaction. For our use case, how should we handle the evaluation of the deadline value? Should we define a default timeout for all mutation operations and use that to set the deadline, or is there another approach we should consider.

**Comment by aakashanandg** [comment]
Revised PR: https://github.com/googleapis/python-spanner/pull/1270
